docs/fundamentals/flow/monitoring-flow.md

```diff
@@ -0,0 +1,89 @@
+(monitoring-flow)=
+# Monitoring Flow
+
+```{caution} 
+The monitoring feature is still in Beta and the API is not stable yet.
+```
+
+A Jina {ref}`Flow <flow-cookbook>` exposes several core metrics that allow you to have a deeper look
+on what is happening inside it. Metrics are particularly useful for building dashboards and alerts, with them, you can, for example, monitor the overall performance 
+of your Flow, detect bottleneck, alert your team when some component of your Flow are down.
+
+Jina Flow expose metrics in the [prometheus format](https://prometheus.io/docs/instrumenting/exposition_formats/) which
+is plain text that is both understandable by human and machine. These metrics are intended to be scrap by [prometheus](https://prometheus.io/). 
+We encourage you to visualize metrics with [grafana](https://grafana.com/)
+
+
+## Using the monitoring in a Flow
+
+A {ref}`Flow <flow-cookbook>` is composed of several Pods namely the Gateway and the Executors (potentially a Head as well). 
+see the {ref}`architecture overview <architecture-overview>` for more details. Each of these Pods are microservices which communicate 
+by gRPC. They will expose their own metrics using the [prometheus client](https://prometheus.io/docs/instrumenting/clientlibs/).
+It means that they are as many metrics endpoints that they are Pods in your Flow. 
+
+Lets gave an example to illustrate it :
+
+````{tab} via Python API
+
+This example shows how to start a Flow with monitoring enable via the Python API:
+
+```python
+from jina import Flow
+
+with Flow(monitoring=True, port_monitoring=9090).add(
+    uses='jinahub://SimpleIndexer', monitoring=True, port_monitoring=9091
+) as f:
+    f.block()
+```
+````
+
+````{tab} via YAML
+This example shows how to start a Flow with monitoring enable via yaml:
+
+In a `flow.yaml` file
+```yaml
+!Flow
+with:
+  monitoring: true
+  port_monitoring: 9090
+executors:
+- uses: jinahub://SimpleIndexer
+  monitoring: true
+  port_monitoring: 9091
```

Comment:
how does this work with replicas then?

Reply:
The same as with the normal port. It will be random